### PR TITLE
Fix wso2-extensions/esb-connector-ldap/issues/24

### PR DESCRIPTION
**Pull request by dilan-premarathna**
Adding search limit, fix for retriving objectGUID attribure and fix for incorrect xml generation
Fixes wso2-extensions/esb-connector-ldap/issues/24

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java [hunk 1]
```diff
@@ -108,6 +136,10 @@ private OMElement prepareNode(SearchResult entityResult, OMFactory factory, OMNa
 
         for (int i = 0; i < returnAttributes.length; i++) {
             attribute = attributes.get(returnAttributes[i]);
+            if (returnAttributes[i].contains(";")) {
```
Comment (Megala21) [inline src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Can we add a comment on why we are doing this?

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java [hunk 2]
```diff
@@ -139,13 +171,34 @@ private SearchResult makeSureOnlyOneMatch(NamingEnumeration<SearchResult> result
         }
     }
 
+    protected byte[] swapBytes(byte[] bytes) {
```
Comment (Megala21) [inline src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Please add a method level comment on what we are doing at this point?

File: src/main/resources/ldap_entry/searchEntry.xml [hunk 3]
```diff
@@ -34,6 +35,7 @@
         <property expression="$func:filters" name="filters" scope="default" type="STRING"/>
         <property expression="$func:dn" name="dn" scope="default" type="STRING"/>
         <property expression="$func:attributes" name="attributes" scope="default" type="STRING"/>
+        <property expression="$func:limit" name="limit" scope="default" type="STRING"/>
```
Comment (Megala21) [inline src/main/resources/ldap_entry/searchEntry.xml]: Can we make this as type INTEGER ?

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java [hunk 4]
```diff
@@ -50,6 +51,17 @@ public void connect(MessageContext messageContext) throws ConnectException {
         String filter = (String) getParameter(messageContext, LDAPConstants.FILTERS);
         String dn = (String) getParameter(messageContext, LDAPConstants.DN);
         String returnAttributes[] = ((String) getParameter(messageContext, LDAPConstants.ATTRIBUTES)).split(",");
+
+        int limit = 0;
+        String searchLimit = (String) getParameter(messageContext, "limit");
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Move "limit" property to the constant file and make use of that

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java [hunk 5]
```diff
@@ -50,6 +51,17 @@ public void connect(MessageContext messageContext) throws ConnectException {
...
+        if (searchLimit != null) {
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Use StringUtils for null check

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java (hunk 1 above)
Comment (dilan-premarathna) [reply src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Done

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java (hunk 2 above)
Comment (dilan-premarathna) [reply src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Done

File: src/main/resources/ldap_entry/searchEntry.xml (hunk 3 above)
Comment (dilan-premarathna) [reply src/main/resources/ldap_entry/searchEntry.xml]: It is not possible to simply change the STRING to INTEGER. The usual practice is to use STRING in similar situations

File: src/main/java/org/wso2/carbon/connector/ldap/LDAPUtils.java [hunk 6]
```diff
@@ -58,6 +58,7 @@ protected static DirContext getDirectoryContext(MessageContext messageContext)
         env.put(Context.PROVIDER_URL, providerUrl);
         env.put(Context.SECURITY_PRINCIPAL, securityPrincipal);
         env.put(Context.SECURITY_CREDENTIALS, securityCredentials);
+        env.put("java.naming.ldap.attributes.binary", LDAPConstants.OBJECT_GUID);
```
[the remark this answers is not in the document]
Comment (dilan-premarathna) [reply src/main/java/org/wso2/carbon/connector/ldap/LDAPUtils.java]: Done

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java (hunk 4 above)
Comment (dilan-premarathna) [reply src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Done

File: src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java (hunk 5 above)
Comment (dilan-premarathna) [reply src/main/java/org/wso2/carbon/connector/ldap/SearchEntry.java]: Done

**Pull request merged**